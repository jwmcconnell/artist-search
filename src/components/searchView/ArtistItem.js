import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const styles = {
  item: {
    border: '1px solid black',
    textAlign: 'center',
    width: '300px',
    height: '200px',
    margin: '5px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center'
  }
};

const ArtistItem = ({ artist }) => {
  return (
    <Link to={`/artist/${artist.name}/${artist.id}`}>
      <li style={styles.item}>
        <h4>{artist.name}</h4>
        <p>{artist.disambiguation}</p>
      </li>
    </Link>
    
  );
};

ArtistItem.propTypes = {
  artist: PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    disambiguation: PropTypes.string
  }).isRequired
};

export default ArtistItem;
