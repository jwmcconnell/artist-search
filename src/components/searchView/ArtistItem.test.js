import React from 'react';
import { shallow } from 'enzyme';
import ArtistItem from './ArtistItem';

describe('ArtistItem component', () => {
  it('renders ArtistItem', () => {
    const wrapper = shallow(<ArtistItem 
      artist={{ name: 'name', id: 'id' }}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
