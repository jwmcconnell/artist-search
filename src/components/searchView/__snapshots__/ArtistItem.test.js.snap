// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistItem component renders ArtistItem 1`] = `
<Link
  to="/artist/name/id"
>
  <li
    style={
      Object {
        "border": "1px solid black",
        "display": "flex",
        "flexDirection": "column",
        "height": "200px",
        "justifyContent": "center",
        "margin": "5px",
        "textAlign": "center",
        "width": "300px",
      }
    }
  >
    <h4>
      name
    </h4>
    <p />
  </li>
</Link>
`;
